lib/client/cluster_client.go

```diff
@@ -563,18 +589,48 @@ func (c *ClusterClient) IssueUserCertsWithMFA(ctx context.Context, params Reissu
 	// target included in it for routing.
 	if !mfaRequired {
 		log.DebugContext(ctx, "MFA not required for access")
+		params.ReusableMFAResponse = nil
+		keyRing, err := certClient.generateUserCerts(ctx, CertCacheKeep, params)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &IssueUserCertsWithMFAResult{
+			KeyRing:     keyRing,
+			MFARequired: proto.MFARequired_MFA_REQUIRED_NO,
+		}, nil
+	}
+
+	if params.ReusableMFAResponse != nil {
+		log.DebugContext(ctx, "MFA is required but reusable MFA response is provided")
```

Comment:
nit: I'd think something is wrong reading this debug log.
```suggestion
		log.DebugContext(ctx, "MFA is required, using reusable MFA response")
```